src/Models/Role.php

```diff
@@ -11,7 +11,8 @@ class Role extends Model
 
     public function users()
     {
-        return $this->belongsToMany(Voyager::modelClass('User'), 'user_roles');
+        return $this->hasMany(Voyager::modelClass('User'))->get()
```

Comment:
I understand why you did it, but this method no longer returns a `Builder` object, so it can't be filtered.  This may not be a problem in this case, but it goes against laravel convention.  i.e. you won't be able to call `$role->users()->where(.....)->get()`

Reply:
Sorry, you are absolutely right, that ->get() should be removed.

Reply:
@fletch3555 I have no idea how to merge the 2 relations and keep the `Builder` object, is that even possible? Please, help!!! :D

If this is impossible, we can use it this way, and fix it when Voyager Upgrade System is released?

Reply:
That's why I started with "I understand why..." haha

It's has to be possible (I'm thinking unions).  Try this: https://laracasts.com/discuss/channels/eloquent/combining-multiple-hasmany-relationships-into-one-model-relation-query-builder

Reply:
@fletch3555 done, with unions, please take a look.
I had to add `->select('*')` and `->select('users.*')`, but we need to improve this... any idea?
for instance, we can't refer to `users` table name directly.

Reply:
I would've just put both `belongsTo...()` calls inline connected with the `union()` instead of other methods.  Also, instead of `->select('*')`, you should be able to do `->getQuery()`([ref](https://laravel.com/api/5.5/Illuminate/Database/Eloquent/Relations/BelongsToMany.html#method_getQuery)), which will return the `Builder` object you need for `->union()`

In short, I picture it something along the lines of this:
```
public function users()
{
return $this->hasMany(Voyager::modelClass('User'))->getQuery()->union($this->belongsToMany(Voyager::modelClass('User'), 'user_roles')->getQuery());
}
```